### ISSUE TITLE
Proposal: changes to prefer-destructuring

### ISSUE THREAD

**Issue by thealjey** [eslint/eslint#7922]
**What rule do you want to change?**
`prefer-destructuring`

I wrote about this in a comment of another issue already.
It was really late and I was tired (sorry for being rude, btw).
But now I have found the time and energy to create a proper issue.

Basically, I think that this rule should only test those parts of the program that create a new variable (`var`, `let`, `const`, function arguments, `import` statements, etc.).
Also, if there's already a destructuring assignment, it might've been written a certain way on purpose.

I think that this rule is a very neat idea, however, it does too much (actually, I'm not even sure it checks function arguments and `import` statements at all; might not be too much, might just be wrong 😃 ).
It produces numerous warnings, that do not help anyone, and most of the time are just false positives.
They make you resort to doing weird things like unnecessarily wrapping your entire statements in parentheses, writing more complicated code than you would need to otherwise just to make a rule pass, or simply ignoring the warnings altogether (which is even worse).

**Does this change cause the rule to produce more or fewer warnings?**
fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
new default behavior

**Please provide some example code that this change will affect:**

```js
global.window = jsdom().defaultView;
const {param} = something.property;
```

**What does the rule currently do for this code?**
warns

**What will the rule do after it's changed?**
nothing

**Comment by not-an-aardvark** [comment]
Closing this issue in favor of https://github.com/eslint/eslint/issues/7886.
